scripts/build_desktop_app_with_firebase.py

```diff
@@ -0,0 +1,281 @@
+      raise ValueError("Prebuilt mac firebase libraries are built for x64 and Release mode only. "
+                       "Please fix the command line arguments and try again")
+
+  if platform.system() == 'Linux':
+    if config == 'Debug':
+      raise ValueError("Prebuilt linux firebase libraries are built with Release mode only. "
+                       "Please fix the --config command line argument and try again.")
+
+def get_vcpkg_triplet(arch, msvc_runtime_library='static'):
+  """ Get vcpkg target triplet (platform definition).
+
+  Args:
+    arch (str): Architecture (eg: 'x86', 'x64').
+    msvc_runtime_library (str): Runtime library for MSVC (eg: 'static', 'dynamic').
+
+  Raises:
+    ValueError: If current OS is not win,mac or linux.
+
+  Returns:
+    (str): Triplet name.
+       Eg: "x64-windows-static".
+  """
+  triplet_name = [arch]
+  if platform.system() == 'Windows':
+    triplet_name.append('windows')
+    triplet_name.append('static')
+    if msvc_runtime_library == 'dynamic':
+      triplet_name.append('md')
+  elif platform.system() == 'Darwin':
+    triplet_name.append('osx')
+  elif platform.system() == 'Linux':
+    triplet_name.append('linux')
+  else:
+    raise ValueError("Unsupported platform. "
+                     "This function works only on Windows, Linux or Mac platforms.")
+
+  triplet_name = '-'.join(triplet_name)
+  print("Using vcpkg triplet: {0}".format(triplet_name))
+  return triplet_name
+
+def build_source_vcpkg_dependencies(sdk_source_dir, arch, msvc_runtime_library):
+  """Build C++ dependencies for firebase source SDK using vcpkg.
+
+  Args:
+   sdk_source_dir (str): Path to Firebase C++ source directory.
+   arch (str): Platform Architecture (example: 'x64').
+   msvc_runtime_library (str): Runtime library for MSVC (eg: 'static', 'dynamic').
+  """
+  # TODO: Remove this once dev branch of firebase-cpp-sdk repo has been merged
+  # onto main branch. This is required because vcpkg lives only in dev branch currently.
+  subprocess.run(['git', 'checkout', 'dev'],
+                 cwd=sdk_source_dir, check=True)
+  subprocess.run(['git', 'pull'], cwd=sdk_source_dir, check=True)
+
+  python_exe = 'python3' if distutils.spawn.find_executable('python3') else 'python'
+  subprocess.run([python_exe, "scripts/gha/install_prereqs_desktop.py"],
+                 cwd=sdk_source_dir, check=True)
+  subprocess.run([python_exe, "scripts/gha/build_desktop.py", "--arch", arch,
+                              "--msvc_runtime_library", msvc_runtime_library,
+                              "--vcpkg_step_only"], cwd=sdk_source_dir, check=True)
+
+def build_app_with_source(app_dir, sdk_source_dir, build_dir, arch,
+                          msvc_runtime_library='static', config=None,
+                          target_format=None):
+  """Build desktop app directly against the firebase C++ SDK source.
+
+  Since this invovles a cmake configure, it is advised to run this on a clean
+  build directory.
```

Comment:
Should a note about this be made at the top of the file / usage printouts?

Reply:
Yup. Makes sense. Easier for users to read this with all the examples at the top. Done.